src/main/java/org/opensearch/knn/index/memory/NativeMemoryCacheManager.java

```diff
@@ -433,9 +433,16 @@ public NativeMemoryAllocation get(NativeMemoryEntryContext<?> nativeMemoryEntryC
                 return result;
             }
         } else {
-            // open graphFile before load
             try (nativeMemoryEntryContext) {
                 String key = nativeMemoryEntryContext.getKey();
+                final NativeMemoryAllocation allocation = cache.getIfPresent(key);
+                // if we already have the allocation we should not open file again as this will cause slowdown in
+                // heavy throughput cases, since open() function do locking while opening and mapping the graph file to
+                // memory.
+                if (allocation != null) {
+                    return allocation;
+                }
+                // open graphFile before load
```

Comment:
Are we able to wrap nativeMemoryEntryContext::load with open call to simplify logic? @Gankris96 @kotwanikunal  was there a reason not to do this?